test/test_isl.py

```diff
@@ -391,6 +391,20 @@ def test_ast_node_list_free():
 
     body.block_get_children()
 
+def test_remove_map_if_callback():
+    umap = isl.UnionMap("{A[0] -> [1]; B[1] -> [2]}")
+
+    umap1 = umap.remove_map_if(lambda m: False)
+    assert umap1 == umap, "map should not change"
+
+    umap2 = umap.remove_map_if(lambda m:
+        m.get_tuple_name(isl.dim_type.in_) == "B")
+    assert umap2 == isl.UnionMap("{A[0] -> [1]}")
+
+    def callback_throws_exception(m):
+        assert False
+    umap3 = umap.remove_map_if(callback_throws_exception)
```

Comment:
@inducer I don't know how to let the test infrastructure to "expect an exception"?

Reply:
https://docs.pytest.org/en/reorganize-docs/new-docs/user/pytest_raises.html